packages/core-web/src/styles/nav-menu-bar.css

```diff
@@ -0,0 +1,39 @@
+.hide-nav-button {
```

Comment:
how about: create a _internal_ base `<overlay>` component, with these styles

use it in `site/page-nav-button` so we can scope these styles still, also to avoid repeating similar logic / template in both

Reply:
Created an `overlay` component for both `<page/site-nav-button>`. Abstracted all the similar template and logic into this component.